buildvertical.targets

```diff
@@ -8,102 +8,92 @@
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
   </PropertyGroup>
   
-  <!-- Add vertical projects to the project group for BuildAllProjects -->
-  <Target Name="AddVerticalProjects"
-          Inputs="%(Project.Identity)"
-          Outputs="fake"
-          DependsOnTargets="DetermineProjectsConfiguration;TrimUnconvertedProjects"
-          BeforeTargets="AnnotateProjectsWithConfiguration">
-    <ItemGroup>
-      <OmitProject Include="%(Project.Identity)" Condition="'%(Project.OmitReason)' != ''">
-        <OmitReason>%(Project.OmitReason)</OmitReason>
-      </OmitProject>
-      <Project Remove="@(OmitProject)" />
-    </ItemGroup>
+ <Target Name="AnnotateProjectsWithConfiguration" 
+         Returns="@(ProjectWithConfiguration)"> 
+    <PropertyGroup>
+      <_ProjectName>$(MSBuildProjectFullPath)</_ProjectName>
+    </PropertyGroup>
+    <Message Importance="Low" Text="$(MSBuildProjectFullPath), Configuration - $(_BestConfiguration)" />
+    <ItemGroup> 
+       <ProjectWithConfiguration Include="$(_ProjectName)" Exclude="@(OmitProject)"> 
```

Comment:
The more I think about it the more I think you should just collapse all the independent targets you are calling down into just this one and take the direct output from FindBestConfiguration task and feed it to ProjectWithConfiguration here. That will eliminate a lot of the duplicated properties and assumptions across these targets as well.